File: pkg/java/src/main/java/dev/openfga/language/validation/AllowableTypesResult.java
package dev.openfga.language.validation;

import java.util.List;

class AllowableTypesResult {
    private final boolean valid;
    private final List<String> allowableTypes;

    public AllowableTypesResult(boolean valid, List<String> allowableTypes) {
        this.valid = valid;
        this.allowableTypes = allowableTypes;
    }

    public boolean isValid() {
        return valid;
    }

    public List<String> getAllowableTypes() {
        return allowableTypes;
    }
}

File: pkg/csharp/src/OpenFga.Language/Validation/AllowableTypesResult.cs
namespace OpenFga.Language.Validation;

internal class AllowableTypesResult {
    public bool Valid { get; }
    public List<string> AllowableTypes { get; }

    public AllowableTypesResult(bool valid, List<string> allowableTypes) {
        Valid = valid;
        AllowableTypes = allowableTypes;
    }
}